### PR TITLE
Bug fix for getting dataframes in TrainingJobAnalytics.

### PR DESCRIPTION
**Pull request by piyushadlakha**
*Issue #, if available:*

*Description of changes:*
Bug fix for getting dataframes in TrainingJobAnalytics.

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have updated the [changelog](https://github.com/aws/sagemaker-python-sdk/blob/master/CHANGELOG.rst) with a description of my changes (if appropriate)
- [ ] I have updated any necessary [documentation](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/sagemaker/analytics.py [hunk 1]
```diff
@@ -246,7 +246,12 @@ def _determine_timeinterval(self):
         """
         description = self._sage_client.describe_training_job(TrainingJobName=self.name)
         start_time = description[u'TrainingStartTime']  # datetime object
-        end_time = description.get(u'TrainingEndTime', datetime.datetime.utcnow())
+        # Incrementing end time by 1 min since cloud watch drops seconds before finding the logs.
```
Comment (laurenyu) [inline src/sagemaker/analytics.py]: s/cloud watch/CloudWatch

File: src/sagemaker/analytics.py [hunk 2]
```diff
@@ -246,7 +246,12 @@ def _determine_timeinterval(self):
...
+        # This results in logs being searched in the time range in which the correct log line was not present.
+        # Example - Log time - 2018-10-22 08:25:55
+        #           Here calculated end time would also be  2018-10-22 08:25:55 (without 1 min addition)
```
Comment (laurenyu) [inline src/sagemaker/analytics.py]: nitpick: there's an extra space after "be"

**Comment by piyushadlakha** [comment]
UT's added.

File: src/sagemaker/analytics.py [hunk 3]
```diff
@@ -246,7 +246,12 @@ def _determine_timeinterval(self):
         """
         description = self._sage_client.describe_training_job(TrainingJobName=self.name)
         start_time = description[u'TrainingStartTime']  # datetime object
-        end_time = description.get(u'TrainingEndTime', datetime.datetime.utcnow())
+        # Incrementing end time by 1 min since CloudWatch drops seconds before finding the logs.
+        # This results in logs being searched in the time range in which the correct log line was not present.
+        # Example - Log time - 2018-10-22 08:25:55
+        #           Here calculated end time would also be 2018-10-22 08:25:55 (without 1 min addition)
+        #           CW will consider end time as 2018-10-22 08:25 and will not be able to search the correct log.
+        end_time = description.get(u'TrainingEndTime', datetime.datetime.utcnow()) + datetime.timedelta(minutes=1)
```
Comment (mvsusp) [inline src/sagemaker/analytics.py]: Please add unit tests for these changes

Comment (piyushadlakha) [reply src/sagemaker/analytics.py]: Updated the change log. UT's were updated in the last commit.

**Pull request merged**